### PR TITLE
[EWS] Link commit URL to pull request

### PR DESCRIPTION
**Pull request by JonWBedard**
#### e09ac790b3a195bc33493f213423da9e5d699d75
<pre>
[EWS] Link commit URL to pull request
<a href="https://bugs.webkit.org/show_bug.cgi?id=236399">https://bugs.webkit.org/show_bug.cgi?id=236399</a>
&lt;rdar://problem/88707207 &gt;

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(GitHub.commit_url): We should link to commit within pull request
(ConfigureBuild.add_pr_details): Pass pr_number to GitHub.commit_url.
* Tools/CISupport/ews-build/steps_unittest.py:
(TestGitHub.test_pr_url_with_repository):
(TestGitHub):
(TestGitHub.test_pr_url_with_invalid_repository):
(TestGitHub.test_commit_url_with_repository):
(TestGitHub.test_commit_url_with_invalid_repository):

Canonical link: <a href="https://commits.webkit.org/247040@main">https://commits.webkit.org/247040@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@289506">https://svn.webkit.org/repository/webkit/trunk@289506</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

File: Tools/CISupport/ews-build/steps.py [hunk 1]
```diff
@@ -91,13 +91,15 @@ def pr_url(cls, pr_number, repository_url=None):
         return '{}/pull/{}'.format(repository_url, pr_number)
 
     @classmethod
-    def commit_url(cls, sha, repository_url=None):
+    def commit_url(cls, sha, repository_url=None, pr_number=None):
```
Comment (aj062) [inline Tools/CISupport/ews-build/steps.py]: Might be a good idea to add unit-test for this

**Review by aj062 — COMMENTED** [review]
Looks good. Can you please share a sample run as well?

File: Tools/CISupport/ews-build/steps.py (hunk 1 above)
Comment (JonWBedard) [reply Tools/CISupport/ews-build/steps.py]: Added tests.

File: Tools/CISupport/ews-build/steps_unittest.py [hunk 2]
```diff
@@ -214,6 +214,26 @@ def behavior(command):
...
 
 
+class TestGitHub(unittest.TestCase):
+    def test_pr_url(self):
+        self.assertEqual(
+            GitHub.pr_url(1234),
+            'https://github.com/WebKit/WebKit/pull/1234',
+        )
+
+    def test_commit_url(self):
```
Comment (aj062) [inline Tools/CISupport/ews-build/steps_unittest.py]: Thanks for adding these. Can also add unit-tests for cases when repository_url was provided (valid and invalid).

**Review by aj062 — APPROVED** [review]

**Comment by JonWBedard** [comment]
> Looks good. Can you please share a sample run as well?

Here is a good example run: https://ews-build.webkit-uat.org/#/builders/9/builds/3269

**Comment by JonWBedard** [comment]
Landed https://commits.webkit.org/247040@main (e09ac790b3a195bc33493f213423da9e5d699d75)!

**Comment by aj062** [comment]
> Here is a good example run: https://ews-build.webkit-uat.org/#/builders/9/builds/3269

I wish I had this example before r+ing. This changes my opinion. 
The [link](https://github.com/WebKit/WebKit/pull/136/commits/3307e2cc7499fcda01b7c89c9788c494a523a7c3) is not at all accessible once PR is landed, while the link without this change applied still works (e.g.: https://github.com/WebKit/WebKit/commit/3307e2cc7499fcda01b7c89c9788c494a523a7c3).

Earlier you mentioned in https://bugs.webkit.org/show_bug.cgi?id=235306#c1 that we wouldn't be able to have the commits displayed in the PR commits tabs after PR is merged. So, even though this url is better when PR is not merged, it's probably better to keep the previous behavior to generate hash urls which works even after PR is merged.

**Comment by JonWBedard** [comment]
> > Here is a good example run: https://ews-build.webkit-uat.org/#/builders/9/builds/3269
> 
> I wish I had seen this example before r+ing. This changes my opinion. The [link](https://github.com/WebKit/WebKit/pull/136/commits/3307e2cc7499fcda01b7c89c9788c494a523a7c3) is not at all accessible once PR is landed, while the link without this change applied still works (e.g.: [3307e2c](https://github.com/WebKit/WebKit/commit/3307e2cc7499fcda01b7c89c9788c494a523a7c3)).
> 
> Earlier you mentioned in https://bugs.webkit.org/show_bug.cgi?id=235306#c1 that we wouldn't be able to have the commits displayed in the PR commits tabs after PR is merged. So, even though this url is better when PR is not merged, it's probably better to keep the previous behavior to generate hash urls which works even after PR is merged.

That's unfortunate behavior, I will revert tomorrow. What's really surprising to me is that these commits seems to stay around as long as PRs reference them in some way, like this one from November: https://github.com/WebKit/WebKit/commit/6001db42e2cd9ff79714fb4c56853259f936d585

**Comment by JonWBedard** [comment]
Reverted in https://commits.webkit.org/247066@main (99429333a1816e6e1da7180e2d0f81d103349ac9)

**Pull request merged**